x-pack/plugins/security_solution/public/common/hooks/use_timeline_events_count.tsx

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { createPortalNode, OutPortal } from 'react-reverse-portal';
+
+/**
+ * A singleton portal for rendering content in the global header
+ */
+const timelineEventsCountPortalNodeSingleton = createPortalNode();
+
+export const useTimelineEventsCountPortal = () => {
+  const [timelineEventsCountPortalNode] = useState(timelineEventsCountPortalNodeSingleton);
+
+  return { timelineEventsCountPortalNode };
```

Comment:
Why don't you return `timelineEventsCountPortalNodeSingleton` ? What the advantage of using `useState` here?

Reply:
that's a good question @cnasikas 
I've just followed the pattern that we're using for other portals

Reply:
I see. I am trying to understand it :). `timelineEventsCountPortalNodeSingleton` is a singleton and after the importation of the file, it will be the same instance across security solution. Theoretically, it shouldn't create rerenders. And I am wondering, is something I do not get with the `useState`?